packages/block-editor/src/components/writing-flow/index.js

```diff
@@ -241,6 +241,14 @@ class WritingFlow extends Component {
 		const hasModifier = isShift || event.ctrlKey || event.altKey || event.metaKey;
 		const isNavEdge = isVertical ? isVerticalEdge : isHorizontalEdge;
 
+		// When presing any key other than up or down, the initial vertical
+		// position must ALWAYS be reset.
```

Comment:
This comment explains _what_ is happening, but not _why_, which would be the more useful context for the future maintainer.